src/qt/notificator.h

```diff
@@ -63,11 +83,15 @@ public slots:
         UserNotificationCenter      /**< Use the 10.8+ User Notification Center (Mac only) */
     };
     QString programName;
-    Mode mode;
+    std::atomic<Mode> mode;
     QSystemTrayIcon *trayIcon;
 #ifdef USE_DBUS
+    QThread *m_dbus_init_thread{nullptr};
+protected:
     QDBusInterface *interface;
+    friend class DBusInitThread;
```

Comment:
`friend` could be avoided if `DBusInitThread` would have `interfaceCreated(QDBusInterface* interface)` signal.

Signal can be invoked from within `run()` using:
```
QTimer::singelShot(0, this, [this, interface]() {
  emit interfaceCreated(interface);
});
```
`singleShot` will detect that `this` (context) lives in different thread and lambda will be invoked in that (receiver) context thread (where `DBusInitThread` was created).

In the result, `std::atomic<Mode> mode;` "atomicity" will not be needed, because 
```
  this->interface = interface;
  mode = Notificator::Freedesktop;
```
..would be performed on thread `Notificator` lives, within a slot (or just lambda) that connected to the `DBusInitThread::interfaceCreated(QDBusInterface*)` signal, and so without *any* race condition, as we are setting two variables, and only one is atomic.

`DBusInitThread` constructor will have to get `QThread*` though for `moveToThread().`. Or jus use `DBusInitThread::thread()` to get that "destination" thread? But that's might break if `interface`-receiving thread is not the same that created `DBusInitThread` object...

Also, adding `interface->setParent(this)` after `this->interface = interface;` would remove need for manual `delete`, unless there's deletion ordering is involved...